drivers/interrupt_controller/intc_gic.c

```diff
@@ -11,19 +11,80 @@
  *       GICv3 interface is not supported.
  */
 
-#include <device.h>
 #include <sw_isr_table.h>
-#include <irq_nextlevel.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <drivers/interrupt_controller/gic.h>
 
 #if CONFIG_GIC_VER >= 3
 #error "GICv3 and above are not supported"
 #endif
 
-struct gic_ictl_config {
-	u32_t isr_table_offset;
-};
+void arm_gic_irq_enable(unsigned int irq)
+{
+	int int_grp, int_off;
+
+	int_grp = irq / 32;
+	int_off = irq % 32;
+
+	sys_write32((1 << int_off), (GICD_ISENABLERn + int_grp * 4));
+}
+
+void arm_gic_irq_disable(unsigned int irq)
+{
+	int int_grp, int_off;
+
+	int_grp = irq / 32;
+	int_off = irq % 32;
+
+	sys_write32((1 << int_off), (GICD_ICENABLERn + int_grp * 4));
+}
+
+int arm_gic_irq_is_enabled(unsigned int irq)
+{
+	int int_grp, int_off;
+	unsigned int enabler;
+
+	int_grp = irq / 32;
+	int_off = irq % 32;
+
+	enabler = sys_read32(GICD_ICENABLERn + int_grp * 4);
+
+	return (enabler & (1 << int_off)) != 0;
+}
+
+void arm_gic_irq_set_priority(
+	unsigned int irq, unsigned int prio, u32_t flags)
+{
+	int int_grp, int_off;
+	u8_t val;
+
+	/* Set priority */
+	sys_write8(prio & 0xff, GICD_IPRIORITYRn + irq);
+
+	/* Set interrupt type */
+	int_grp = irq / 4;
+	int_off = (irq % 16) * 2;
+
+	val = sys_read8(GICD_ICFGRn + int_grp);
+	val &= ~(GICC_ICFGR_MASK << int_off);
+	if (flags & IRQ_TYPE_EDGE)
```

Comment:
zephyr coding style requires braces even for one-liners

Reply:
Right, I will fix all one-liners without braces in this file while I am at it.